department-of-veterans-affairs/caseflow
# Fix state dropdown bugs on Add Claimant Page

Resolves [CASEFLOW-1337](https://vajira.max.gov/browse/CASEFLOW-1337)

### Description
Fix bugs in state dropdown on Add Claimant Page

### Acceptance Criteria
- [ ] State field is clearable in form
- [ ] 'Continue to next step' is enabled if all required fields after state are valid
- [ ] Search upon user type is not off by one  when returning search results

### Testing Plan
1. Switch to a user with intake privileges
2. Search for a veteran file number
3. Select supplemental claim
4. Complete form  but save state dropdown for last
  a. Type character that matches the first letter of multiple states e.g. I, N, C, M
5. Verify that options appear in alphabetical order
6. Verify that clicking x in dropdown clears field
7. Select any state
8. Verify that continue button enables


### User Facing Changes

 BEFORE|AFTER
<img width="690" alt="Screen Shot 2021-04-19 at 4 55 05 PM" src="https://user-images.githubusercontent.com/68879427/115317280-03d93200-a130-11eb-9230-89c7247db158.png">|<img width="660" alt="Screen Shot 2021-04-19 at 4 52 50 PM" src="https://user-images.githubusercontent.com/68879427/115317237-e4420980-a12f-11eb-9fa2-e7f37c758ff4.png">
